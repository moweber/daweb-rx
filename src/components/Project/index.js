import React, { useState } from "react";

const Project = () => {
  const [projects] = useState([
    {
      id: 0,
      name: "React Pathway Portfolio",
      description:
        "This project explored the basics of building a React app and deploying it to GitHub pages.",
      liveLink: "https://moweber.github.io/photo-port/",
      srcLink: "https://github.com/moweber/photo-port",
    },
    {
      id: 1,
      name: "Oh Snap",
      description:
        "This was a practice project created while learning React concepts.",
      liveLink: "https://moweber.github.io/photo-port/",
      srcLink: "https://github.com/moweber/photo-port",
    },
    {
      id: 2,
      name: "PROJECT NAME",
      description: "PROJECT DESCRIPTION",
      liveLink: "https://moweber.github.io/photo-port/",
      srcLink: "https://github.com/moweber/photo-port",
    },
    {
      id: 3,
      name: "PROJECT NAME",
      description: "PROJECT DESCRIPTION",
      liveLink: "https://moweber.github.io/photo-port/",
      srcLink: "https://github.com/moweber/photo-port",
    },
    {
      id: 4,
      name: "PROJECT NAME",
      description: "PROJECT DESCRIPTION",
      liveLink: "https://moweber.github.io/photo-port/",
      srcLink: "https://github.com/moweber/photo-port",
    },
    {
      id: 5,
      name: "PROJECT NAME",
      description: "PROJECT DESCRIPTION",
      liveLink: "https://moweber.github.io/photo-port/",
      srcLink: "https://github.com/moweber/photo-port",
    },
  ]);

  return (
    <div className="mPortfolio">
      {projects.map((project, i) => (
        <div
          key={project.id}
          style={{
            backgroundImage: `url(${require(`../../assets/projects/${i}.png`)})`,
            WebkitBackgroundSize: "100% auto",
          }}
          className="mProject"
        >
          <div className="mProjectLinks">
            <h2>
              <a target="_blank" rel="noreferrer" href={project.liveLink}>
                {project.name}
              </a>
              <a
                href={project.srcLink}
                target="_blank"
                rel="noreferrer"
                aria-label="GitHub Icon"
                className="fa fa-github mProjectSrcLink"
              >
                {" "}
              </a>
            </h2>
            <h5>{project.description}</h5>
          </div>
        </div>
      ))}
    </div>
  );
};

export default Project;
