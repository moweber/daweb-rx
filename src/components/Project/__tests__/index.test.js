import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Project from "..";

afterEach(cleanup);

describe("Project is rendering", () => {
  it("renders", () => {
    render(<Project />);
  });

  it("renders", () => {
    const { asFragment } = render(<Project />);
    expect(asFragment()).toMatchSnapshot();
  });
});
