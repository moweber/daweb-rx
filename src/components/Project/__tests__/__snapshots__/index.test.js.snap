// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project is rendering renders 1`] = `
<DocumentFragment>
  <div
    class="mPortfolio"
  >
    <div
      class="mProject"
      style="background-image: url(0.png);"
    >
      <div
        class="mProjectLinks"
      >
        <h2>
          <a
            href="https://moweber.github.io/photo-port/"
            rel="noreferrer"
            target="_blank"
          >
            PROJECT NAME
          </a>
          <a
            aria-label="GitHub Icon"
            class="fa fa-github mProjectSrcLink"
            href="https://github.com/moweber/photo-port"
            rel="noreferrer"
            target="_blank"
          >
             
          </a>
        </h2>
        <h5>
          PROJECT DESCRIPTION
        </h5>
      </div>
    </div>
    <div
      class="mProject"
      style="background-image: url(1.png);"
    >
      <div
        class="mProjectLinks"
      >
        <h2>
          <a
            href="https://moweber.github.io/photo-port/"
            rel="noreferrer"
            target="_blank"
          >
            PROJECT NAME
          </a>
          <a
            aria-label="GitHub Icon"
            class="fa fa-github mProjectSrcLink"
            href="https://github.com/moweber/photo-port"
            rel="noreferrer"
            target="_blank"
          >
             
          </a>
        </h2>
        <h5>
          PROJECT DESCRIPTION
        </h5>
      </div>
    </div>
    <div
      class="mProject"
      style="background-image: url(2.png);"
    >
      <div
        class="mProjectLinks"
      >
        <h2>
          <a
            href="https://moweber.github.io/photo-port/"
            rel="noreferrer"
            target="_blank"
          >
            PROJECT NAME
          </a>
          <a
            aria-label="GitHub Icon"
            class="fa fa-github mProjectSrcLink"
            href="https://github.com/moweber/photo-port"
            rel="noreferrer"
            target="_blank"
          >
             
          </a>
        </h2>
        <h5>
          PROJECT DESCRIPTION
        </h5>
      </div>
    </div>
    <div
      class="mProject"
      style="background-image: url(3.png);"
    >
      <div
        class="mProjectLinks"
      >
        <h2>
          <a
            href="https://moweber.github.io/photo-port/"
            rel="noreferrer"
            target="_blank"
          >
            PROJECT NAME
          </a>
          <a
            aria-label="GitHub Icon"
            class="fa fa-github mProjectSrcLink"
            href="https://github.com/moweber/photo-port"
            rel="noreferrer"
            target="_blank"
          >
             
          </a>
        </h2>
        <h5>
          PROJECT DESCRIPTION
        </h5>
      </div>
    </div>
    <div
      class="mProject"
      style="background-image: url(4.png);"
    >
      <div
        class="mProjectLinks"
      >
        <h2>
          <a
            href="https://moweber.github.io/photo-port/"
            rel="noreferrer"
            target="_blank"
          >
            PROJECT NAME
          </a>
          <a
            aria-label="GitHub Icon"
            class="fa fa-github mProjectSrcLink"
            href="https://github.com/moweber/photo-port"
            rel="noreferrer"
            target="_blank"
          >
             
          </a>
        </h2>
        <h5>
          PROJECT DESCRIPTION
        </h5>
      </div>
    </div>
    <div
      class="mProject"
      style="background-image: url(5.png);"
    >
      <div
        class="mProjectLinks"
      >
        <h2>
          <a
            href="https://moweber.github.io/photo-port/"
            rel="noreferrer"
            target="_blank"
          >
            PROJECT NAME
          </a>
          <a
            aria-label="GitHub Icon"
            class="fa fa-github mProjectSrcLink"
            href="https://github.com/moweber/photo-port"
            rel="noreferrer"
            target="_blank"
          >
             
          </a>
        </h2>
        <h5>
          PROJECT DESCRIPTION
        </h5>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
